op-challenger/fault/game.go

```diff
@@ -0,0 +1,120 @@
+package fault
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-challenger/config"
+	"github.com/ethereum-optimism/optimism/op-challenger/fault/alphabet"
+	"github.com/ethereum-optimism/optimism/op-challenger/fault/cannon"
+	"github.com/ethereum-optimism/optimism/op-challenger/fault/types"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type Actor interface {
+	Act(ctx context.Context) error
+}
+
+type GameInfo interface {
+	GetGameStatus(context.Context) (types.GameStatus, error)
+	LogGameInfo(ctx context.Context)
+}
+
+type Game struct {
+	agent                   Actor
+	agreeWithProposedOutput bool
+	caller                  GameInfo
+	logger                  log.Logger
+}
```

Comment:
This is a bit confusing since we already have the notion of a `Game` inside `fault/types/game.go`. Once is representative of the challenger components involved in responding to a dispute game, and the other (types/game.go) is a state representation of the fault dispute game.

A lot is being done in this PR. I think it would behove us to talk a step back and put together concrete, small and specific tickets and steps to progress the challenger instead of large(r >600 diff) prs.

That said, this does bring us further toward our goal and i quite strongly dislike blocking progress so I think we can work with what is here.

Reply:
Yeah the naming is getting really hard - we already have a `GameAgent` which is potentially what I'd have called this. It could be `GamePlayer` I guess, or I'm open to any suggestions you have for better naming.

In terms of the size of the PR I don't think it's as big as you think - there's a bunch of code that got moved but is otherwise unchanged. That makes the diff look big but the functional change is fairly small - just wiring in the existing dispute game loader and having it create an instance of `Game` for each game it finds (what is now `Game` was just the single `service` instance previously).